packages/babel-core/test/config-chain.js

```diff
@@ -1177,6 +1182,23 @@ describe("buildConfigChain", function () {
         );
       });
 
+      nodeGte23("should load babel.config.mjs synchronously", async () => {
```

Comment:
Note that Node.js has plans to backport `require(esm)` unflagged support to Node.js 22, after letting it bake for a while in Node.js 23.

Reply:
OK, we will update the helper function when the backport is released. 

Aside: should we reconsider the current worker/client approach in babel/{eslint-parser,register} now that `require(esm)` is a thing?

Reply:
> should we reconsider the current worker/client approach in babel/{eslint-parser,register} now that require(esm) is a thing?

Probably yes? Maybe keep it as a fallback for Node.js 18&20 (if `require(esm)` is not backported to them), but default to in-thread transform.

Reply:
Could you squash all the other commits into one? :)

So that then we merge this PR as two separate commits.